src/artm/core/dictionary.h

```diff
@@ -46,7 +46,7 @@ class Dictionary {
   std::vector<DictionaryEntry> entries_;
   std::unordered_map<Token, int, TokenHasher> token_index_;
   std::unordered_map<int, Token> index_token_;
-  std::unordered_map<int, std::unordered_map<int, int> > cooc_values_;
+  std::unordered_map<int, std::unordered_map<int, std::vector<float> > > cooc_values_;
```

Comment:
Well, different formulas of coherence require different values, including collection characteristics. It seems to be incorrect to limit user with only described formula. Anya and I decided to implement several modes in coherency plugin with different formulas. So I think, it's necessary to collect such values, as total_items_count and entry.items_count

Maybe we can ask for dictionary file, collected during parsing, as a third parameter of parsing in cooc mode and add necessary data from file into it. The only limit is that user must guarantee that numbers of tokens in dictionary and vocab file should be equal and tokens must have the same order. Is this way too complex?

Regards, Murat.

07.07.2015, 00:31, "sashafrey" <notifications@github.com>:
> In src/artm/core/dictionary.h:
>
>> @@ -46,7 +46,7 @@ class Dictionary { > std::vector<DictionaryEntry> entries_; > std::unordered_map<Token, int, TokenHasher> token_index_; > std::unordered_map<int, Token> index_token_; > - std::unordered_map<int, std::unordered_map<int, int> > cooc_values_; > + std::unordered_map<int, std::unordered_map<int, std::vector<float> > > cooc_values_;
>
> Not if we say that cooc file contains PMI. Then you simply count coherence as described in https://s3-eu-west-1.amazonaws.com/artm/voron15slds.pdf (page 7, last paragraph).
>
> coh = 2/(k * (k-1)) sum_{i, j} PMI(w_i, w_j) (1)
>
> The value (last column) in cooc file needs to be PMI(w_i, w_j), and BigARTM do no further processing of this number. However, BigARTM does not care if the user have actually calculated PMI or not -- we just calculate via a simple formula (1).
>
> Does this solve the issue, or I'm still missing something important?
>
> —
> Reply to this email directly or view it on GitHub.